src/wrdesc.c

```diff
@@ -15,16 +15,16 @@
  * If an array overflow occurs, then an appropriate error message
  * will be written via bort().
  *
- * @param desc - bit-wise representation of descriptor to be written
- * into descary.
- * @param descary - array of descriptors.
- * @param ndescary - number of descriptors written so far into
+ * @param desc - WMO bit-wise representation of descriptor to be
+ * written into descary.
+ * @param descary - Array of descriptors.
+ * @param ndescary - Number of descriptors written so far into
  * descary.
  *
  * @author J. Ator @date 2004-08-18
  */
 void
-wrdesc(f77int desc, f77int *descary, f77int *ndescary)
+wrdesc(int desc, int *descary, int *ndescary)
```

Comment:
How come we are changing these? Is there an issue to describe this change?

Reply:
This f77int typedef is a remnant from back when we had multiple _4, _8, etc. builds of the library, as a way to easily link a C "int" to a Fortran integer (in the case of the _4 build) vs. linking a "long" to a Fortran integer (in the case of the _8 build).  That way, we could use a single typedef throughout the C source code that would work for either case.  But now, following the resolution of #78 and #257, there's only a single _4 library build now, and so f77int is always an "int" and therefore the typedef is superfluous and can go away.

So, in a nutshell, this is basically remaining cleanup from the consolidation down to a single _4 library build, but on the C side which we're still trying to get cleaned up and thereby simplify the code and make it cleaner and easier to understand.

And to answer your earlier question about why we need to change these C function signatures in the first place, that goes back to #79 where we're trying to, among other things, "remove the UNDERSCORE and all related shenanigans" (your words ;-).  Modernizing this C-Fortran interface using BIND also means that we no longer need to pass extra string length arguments when calling Fortran from C, which is yet another change to many of these function signatures.  And while doing so I'm also following the earlier model that Ron M. laid out where you append _f when calling Fortran from C, and _c when calling C from Fortran.  So just trying to keep everything consistent throughout the library.

Reply:
OK, makes sense.